woocommerce/woocommerce-gutenberg-products-block
# Create `withProduct` HOC

This PR includes several changes:
* Moves `getProduct()` to a HOC following @nerrad's suggestion for the `<ProductsControl>` component. This way, we will be able to reuse `getProduct()` in _Reviews by Product_ without duplicating the logic.
* Converts `<FeaturedProduct>` to a functional component.
* Removes the `retrying` state from `ApiErrorPlaceholder` and make it rely on a `isLoading` prop, this way we don't have to maintain the same state in two different components.
* Moves the error messages to `ApiErrorPlaceholder`.

### How to test the changes in this Pull Request:

To test, basically add a _Featured Product_ block and verify it still works as expected.

**[Bonus points]** Test that errors are still displayed:

1. Replace [this line](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/src/RestApi/Controllers/Products.php#L104) with:
```diff
-return true;
+return new \WP_Error( 'DERP', 'You cannot read products.' );
``` 
2. Add a _Featured Product_ block.
3. Notice the error appears as expected.
